lib/evmone/eof.cpp

```diff
@@ -448,37 +455,41 @@ std::variant<EOFValidationError, int32_t> validate_max_stack_height(
         if (opcode == OP_CALLF)
         {
             const auto fid = read_uint16_be(&code[i + 1]);
+            if (fid >= header.get_type_count())
```

Comment:
This is a new check. Why is it needed now?

Reply:
I removed the addition of those checks. Turns out the check I replaced in validate_instructions was not strict enough. Now it is and all test pass without the additional check.